features/step_definitions/metering.rb

```diff
@@ -225,3 +224,47 @@
   res = host.exec(shell_cmd)
   raise "#{cb.metering_namespace} #{op} unsuccessfully" unless res[:success]
 end
+
+# install metering via OLM,
+Given /^the#{OPT_QUOTED} metering service is installed(?: to $QUOTED)? using OLM$/ do | metering_ns |
+  ensure_admin_tagged
+  # 1. create the metering namespace
+  metering_ns ||= "openshift-metering"
+  project(metering_ns)
+  step %Q/I switch to cluster admin pseudo user/
+  @result = user.cli_exec(:create_namespace, name: metering_ns)
+  # prep for OLM need to define configs via YAML
+  # 1. metering-catalogsourceconfig.yaml
+  catalog_source_config = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/metering/OLM_examples/metering-catalogsourceconfig.yaml"
+  # 2. metering-operatorgroup.yaml
+  operator_group = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/metering/OLM_examples/metering-operatorgroup.yaml"
+  # 3. metering-subscription.yaml
+  subscription = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/metering/OLM_examples/metering-subscription.yaml"
+  metering_crd = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/metering/OLM_examples/metering_crd.yaml"
+
+  # should really get the namespace dynamically
+  @result = user.cli_exec(:apply, f: catalog_source_config, n: 'openshift-marketplace')
+  @result = user.cli_exec(:apply, f: operator_group, n: project.name)
+  @result = user.cli_exec(:apply, f: subscription, n: project.name)
+  # wait for two initial metering pods to become running first and then apply CRD
+  step %Q/a pod becomes ready with labels:/, table(%{
+    | app=metering-operator |
+  })
+
+  step %Q/a pod becomes ready with labels:/, table(%{
+    | olm.catalogSource=metering |
+  })
+  @result = user.cli_exec(:apply, f: metering_crd, n: project.name)
+  raise "OLM install of metering failed" unless @result[:success]
+end
+
+# XXX: currently OLM uninstall is TBD, we uninstall by removing the namespace
+Given /^the#{OPT_QUOTED} metering service is uninstalled using OLM$/ do | metering_ns |
```

Comment:
I don't see this step registered in cleanup. Or did I miss that?

Reply:
@akostadinov, the general usage from `metering` is that you install it once and run the various test.  We shouldn't be installing and uninstalling it for every test feature.  If we want to uninstall `metering`, the user should call it explicitly

Reply:
It is preferable to have things pre-installed on cluster install instead of whenever a test runs. If we keep it like this, then making step ensure_destructive_tagged is needed to avoid installing this at the same time by two scenarios.

Another issue is in case you want at some point to insall with a different configuration. How would you handle it?

Reply:
good point, I'll add that tag.  WRT the configuration...the OLM install is still in its infancy, so this step will definitely need to be modified.  How about default to these for now and perhaps add a table option later to let user overwrite the defaults?

Reply:
we need to make sure to tidy things up after GA

Reply:
ok, will do.  TBH, metering tests are generally run after the initial installation is done unless the testcase is installation specific.

Reply:
@akostadinov PR updated with tags and added more sanity check.  PTAL.  Thanks!